### PR TITLE
Add channels entries for image architecture

### PR DESCRIPTION
**Pull request by hakman**
This should fix most requests around having correct images for new clusters and upgrades.

Fixes: #11023

/cc @rifelpet @olemarkus @bharath-123

### CODE REVIEW

File: channels/stable [hunk 1]
```diff
@@ -3,72 +3,94 @@ spec:
...
       kubernetesVersion: ">=1.17.0 <1.18.0"
     - name: 099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20210119.1
       providerID: aws
+      architectureID: amd64
       kubernetesVersion: ">=1.18.0"
-    - providerID: gce
+    - name: 099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-arm64-server-20210119.1
+      providerID: aws
+      architectureID: arm64
+      kubernetesVersion: ">=1.18.0"
```
Comment (rifelpet) [inline channels/stable:68]: At the end of `pkg/apis/kops/channel.go`'s `FindImage` function we log a warning if multiple matches are found. 

This means that as soon as this PR is merged, all existing releases of kops will begin emitting a warning on k8s clusters >= k8s 1.18.0. Perhaps we only add an arm64 image starting with the k8s version associated with the kops version we anticipate this being released in?

Comment (hakman) [reply channels/stable]: Agreed. Should be fine now.

**Comment by bharath-123** [comment]
Tried testing this out with the toolbox command, got this error
```
bharath12345@Bharaths-MacBook-Pro kops % kops toolbox instance-selector nodes --vcpus 4 --flexible --usage-class spot --cpu-architecture arm64 --cluster-autoscaler --dry-run
Using cluster from kubectl context: test-stuff.kops-cluster.com

I0314 18:48:18.623902   99607 populate_instancegroup_spec.go:260] Cannot set default Image for CloudProvider="aws"

unable to determine default image for InstanceGroup nodes
```

File: pkg/apis/kops/channel.go [hunk 2]
```diff
@@ -259,13 +262,16 @@ const (
...
+func (c *Channel) FindImage(provider CloudProviderID, kubernetesVersion semver.Version, architecture architectures.Architecture) *ChannelImageSpec {
 	var matches []*ChannelImageSpec
 
 	for _, image := range c.Spec.Images {
 		if image.ProviderID != string(provider) {
 			continue
 		}
+		if image.ArchitectureID != string(architecture) {
+			continue
+		}
```
Comment (rifelpet) [inline pkg/apis/kops/channel.go:274]: I'm wondering if we shouldn't `continue` if `image.ArchitectureID != ""`. I dont know the workflows that users have for hosting their own channels files but if they dont copy the associated architecture additions in this PR when upgrading to this version of kops then they'll no longer find any images. Perhaps a release note mentioning the change for offline installations would be sufficient.

**Comment by rifelpet** [comment]
@bharath-123 your kops build is likely using the upstream master channels file which doesn't have the architecture additions in it. If you update your cluster spec's `channel` field to the file's URL from this PR then it should work.

https://raw.githubusercontent.com/kubernetes/kops/a5f5f984c99a9cf6b31ab87beee15a5248f50107/channels/stable

File: pkg/apis/kops/channel.go (hunk 2 above)
Comment (hakman) [reply pkg/apis/kops/channel.go:274]: I think for now we should go with the release note. We can see if later we need to improve.

**Comment by bharath-123** [comment]
> @bharath-123 your kops build is likely using the upstream master channels file which doesn't have the architecture additions in it. If you update your cluster spec's `channel` field to the file's URL from this PR then it should work.
> 
> https://raw.githubusercontent.com/kubernetes/kops/a5f5f984c99a9cf6b31ab87beee15a5248f50107/channels/stable

Got it @rifelpet Thanks for clearing that out!

File: pkg/util/templater/template_functions.go [hunk 3]
```diff
@@ -63,9 +64,9 @@ func (r *Templater) templateFuncsMap(tm *template.Template) template.FuncMap {
 
 	}
 
-	funcs["ChannelRecommendedImage"] = func(cloud, k8sVersion string) string {
+	funcs["ChannelRecommendedImage"] = func(cloud, k8sVersion string, architecture architectures.Architecture) string {
```
Comment (rifelpet) [inline pkg/util/templater/template_functions.go:67]: normally changing a template function's signature would be worthy of a release note but this hasnt been in a stable release yet so we should be fine. we do need to update the docs that use this function though https://github.com/kubernetes/kops/blob/master/docs/operations/cluster_template.md

Comment (hakman) [reply pkg/util/templater/template_functions.go:67]: Done.

**Review by rifelpet — APPROVED** [review]

**Comment by yurrriq** [comment]
I'm running into issues using kOps 1.20.0: `wrong type for value; expected architectures.Architecture; got string`

using this `snippets/common/ami`:
```
{{ default (ChannelRecommendedImage $.cloud (include "kubernetesVersion" $) $.architecture) $.image }}
```
snippets/common/kubernetesVersion:
```
{{ default ChannelRecommendedKopsKubernetesVersion $.kubernetesVersion }}
```

with this template
```
{{ $image := include "ami" $) }}
```

and these values:
```yaml
architecture: amd
cloud: aws
```

**Pull request merged**